model_to_project_task/project.py

```diff
@@ -0,0 +1,99 @@
+# coding: utf-8
+# © 2015 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api, _, SUPERUSER_ID
+from openerp.exceptions import Warning as UserError
+from openerp.tools.safe_eval import safe_eval
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    @api.model
+    def _authorised_models(self):
+        """ similar in new api to
+            openerp.addons.base.res.res_request import referencable_models
+
+            Inherit this method to add more models depending of your
+            modules dependencies
+        """
+        models = self.env['ir.model'].search([])
+        return [(x.model, x.name) for x in models]
+
+    action_id = fields.Many2one(
+        'ir.actions.act_window', string="Action",
+        help="Action called to go to the original window")
+    reference = fields.Reference(
+        selection='_authorised_models', string="Task Origin", readonly=True)
+
+    @api.model
+    def default_get(self, fields):
+        vals = super(ProjectTask, self).default_get(fields)
+        vals['partner_id'] = (
+            self.env['res.users'].browse(self.env.uid).partner_id.id)
+        vals['user_id'] = False
```

Comment:
These two default are opinionated and can interfere with other workflows.
Also, and empty `user_id` can be a problem when the user can only see followed Tasks, since it can prevent the user to be automatically added to the follower list.